gk/main.c

```diff
@@ -2094,6 +2100,32 @@ update_flow_entry(struct flow_entry *fe, struct ggu_policy *policy)
 	}
 }
 
+static void
+update_flow_table(struct gk_fib *fib, struct ggu_policy *policy,
+	struct gk_instance *instance, struct gk_config *gk_conf,
+	uint32_t rss_hash_val)
+{
+	struct flow_entry *fe;
+
+	if (fib == NULL || fib->action != GK_FWD_GRANTOR) {
+		/* Drop this solicitation to add a policy decision. */
+		char err_msg[128];
+		int ret = snprintf(err_msg, sizeof(err_msg),
+			"gk: at %s initialize flow entry error",
+			__func__);
+		RTE_VERIFY(ret > 0 && ret < (int)sizeof(err_msg));
+		print_flow_err_msg(&policy->flow, err_msg);
+		return;
+	}
+
+	fe = add_new_flow_from_policy(fib, policy,
```

Comment:
It seems that `update_flow_table()` is the only caller of `add_new_flow_from_policy()`. Moreover, these functions share the same intent, and `add_new_flow_from_policy()` is small. It would be better to bring the code of `add_new_flow_from_policy()` into `update_flow_table()` and to drop `add_new_flow_from_policy()`.